### ISSUE TITLE
Re: blend skelleton. different number of bones.

### ISSUE THREAD

**Issue by overvolting** [Unvanquished/Unvanquished#584]
sometimes when I'm in /devmap with noclip I get flooded with spam messages and can't read the chat. 
it says Re: blend skelleton, different number of bones. 38 != 256.

It's terrible!

**Comment by Viech** [comment]
Interesting. First of all, @gimhael, is it possible that your bone limit increase to 256 (e3dcb6a) has anything to do with it? As far as I remember, all our models have less bones than 256 and only the Spiker has more bones than the old limit of 128. @overvolting, does a spiker exist on the map when this happens?

It should also be noted that this looks similiar to #503.

**Comment by Kangz** [comment]
@Viech it doesn't seem related to that because 38 != 128 too, also I'm not in favor of keeping the increased numbre of bones.

**Comment by DolceTriade** [comment]
If there is an error building the skeleton, I think numBones is assigned MAX_BONES

**Comment by Viech** [comment]
:neutral_face:

**Comment by VReaperV** [comment]
This does not seem to happen anymore.

**Comment by slipher** [comment]
Yes it does. It's pretty easy to reproduce if you just go on spec and scroll back and forth between different players.

**Comment by VReaperV** [comment]
> Yes it does. It's pretty easy to reproduce if you just go on spec and scroll back and forth between different players.

No, that is a different issue. Warn caused by switching between players/bots is at https://github.com/Unvanquished/Unvanquished/blob/675e916db5a57ffd07cbbfa68f192a184a9074c1/src/cgame/cg_players.cpp#L2980, `RE_BlendSkeleton` warning is caused by https://github.com/DaemonEngine/Daemon/blob/ad878611618a7752698f7359b766b0890bd1fd16/src/engine/renderer/tr_animation.cpp#L1486.

**Comment by slipher** [comment]
I just tried it and I got the message like the OP:

```
^3Warn: trap_R_BlendSkeleton: different number of bones 38 != 122
^3Warn: trap_R_BlendSkeleton: different number of bones 38 != 122
^3Warn: trap_R_BlendSkeleton: different number of bones 38 != 122
```

I've definitely seen the skipnotify variant too though. I guess there are two variants of the bug.

**Comment by illwieckz** [comment]
I suspect the spectating player switching bug comes from the fact we may play an animation from the previous models with the new model, or the other way, an animation from the new model with the old model.

**Comment by VReaperV** [comment]
Intereseting, I'm pretty sure I hadn't seen that specific warn before, but I've been able to get it now locally as well, so I was wrong about it earlier.

**Comment by VReaperV** [comment]
> I suspect the spectating player switching bug comes from the fact we may play an animation from the previous models with the new model, or the other way, an animation from the new model with the old model.

I believe so too. That code is unfortunately very messy, but hopefully I can clean it up in #3276.
